src/js/components/projects/projectForm.tsx

```diff
@@ -0,0 +1,101 @@
+/* eslint-disable @typescript-eslint/camelcase */
+import Button from '@salesforce/design-system-react/components/button';
+import Input from '@salesforce/design-system-react/components/input';
+import Textarea from '@salesforce/design-system-react/components/textarea';
+import i18n from 'i18next';
+import React, { useState } from 'react';
+import { connect } from 'react-redux';
+import { Redirect, withRouter } from 'react-router-dom';
+
+import { ObjectsActionType, postObject } from '@/store/actions';
+import { OBJECT_TYPES } from '@/utils/constants';
+import routes from '@/utils/routes';
+
+interface Props {
+  productName: string;
+  productSlug: string;
+  doPostObject: ObjectsActionType;
+}
+
+const ProjectForm = ({ productName, productSlug, doPostObject }: Props) => {
+  const [description, setDescription] = useState('');
+  const [name, setName] = useState('');
+  const [projectCreateActive, setprojectCreateActive] = useState(false);
+  const [hasName, setHasName] = useState(true);
+
+  const handleSubmit = () => {
+    if (name === '' && hasName) {
+      setHasName(false);
+      return;
+    }
```

Comment:
I'm not clear on this `setHasName` logic throughout. It seems odd to me that the first time I try to submit without a name I get a warning, but then on subsequent times it just goes through and tries anyway. I'd say let's either prevent consistently, or allow the server-side validation to do its thing.